solutions/devsprint-riccardo-1/OnboardingChallenge/Protocols/ViewCode.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  ViewCode.swift
+//  OnboardingChallenge
+//
+//  Created by Pedro Gabriel on 15/12/21.
+//
+
+import UIKit
+
+protocol ViewCode where Self: UIView {
```

Comment:
In this line, the protocol is constrained to `UIView` meaning that only `UIView` can inherit from this protocol.

Could you explain why you make this decision? For example, I could see that a `UIViewController` may benefit from the same methods. And if tomorrow Apple introduces another way to define views with autolayout, they should be able to inherit from the protocol as well.

Reply:
It's a good question!
But I think that the common Design System conception / usage in this case are elements inside an UIView, so, even you have some button, or some complex UI (compositions), you use UIViews as base class.
Maybe @rdgborges can give another perception about that.
But make sense don't enforce this usage restricted only to UIViews.